File: backend_js/routes/companies.js
const express = require('express')
const router = express.Router()
const admin = require('../firebase-admin')

async function verifyToken(req, res, next) {
  const authHeader = req.headers.authorization
  if (!authHeader?.startsWith('Bearer ')) return res.status(401).json({ error: 'Unauthorized' })
  const token = authHeader.split('Bearer ')[1]
  try {
    const decoded = await admin.auth().verifyIdToken(token)
    req.uid = decoded.uid
    req.email = decoded.email
    next()
  } catch {
    res.status(401).json({ error: 'Invalid token' })
  }
}

// POST /api/companies/profile — save or update company profile
router.post('/profile', verifyToken, async (req, res) => {
  try {
    const db = admin.firestore()
    // Exclude logo (base64) to stay under Firestore 1MB doc limit
    const { logo, ...companyData } = req.body
    const data = {
      ...companyData,
      uid: req.uid,
      email: req.email,
      updatedAt: admin.firestore.FieldValue.serverTimestamp(),
    }
    await db.collection('companies').doc(req.uid).set(data, { merge: true })
    res.json({ success: true, message: 'Company profile saved' })
  } catch (err) {
    console.error('Error saving company:', err)
    res.status(500).json({ error: 'Failed to save company profile' })
  }
})

// GET /api/companies/profile — fetch company profile
router.get('/profile', verifyToken, async (req, res) => {
  try {
    const db = admin.firestore()
    const doc = await db.collection('companies').doc(req.uid).get()
    if (!doc.exists) return res.json({ profile: null })
    res.json({ profile: doc.data() })
  } catch (err) {
    console.error('Error fetching company:', err)
    res.status(500).json({ error: 'Failed to fetch company profile' })
  }
})

module.exports = router

File: backend/api/routes/companies.py
"""
api/routes/companies.py
-----------------------
Endpoints for recruiter/company profile management.

Routes:
  POST /api/companies/profile — save/update company profile
  GET  /api/companies/profile — fetch own company profile
"""
from fastapi import APIRouter, Depends

from core.firebase import firebase_db
from core.security import get_current_user

router = APIRouter(prefix="/companies", tags=["Companies"])


@router.post("/profile", summary="Save or update company profile")
async def save_company_profile(body: dict, user: dict = Depends(get_current_user)):
    """Saves the authenticated recruiter's company profile to Firestore.

    Note: The 'logo' field (base64) is excluded to stay under Firestore's 1MB limit.
    """
    db = firebase_db()
    # Exclude logo (base64) to stay under Firestore 1MB doc limit
    body.pop("logo", None)

    data = {
        **body,
        "uid": user["uid"],
        "email": user["email"],
        "updatedAt": firebase_db.SERVER_TIMESTAMP,
    }
    db.collection("companies").document(user["uid"]).set(data, merge=True)
    return {"success": True, "message": "Company profile saved"}


@router.get("/profile", summary="Get own company profile")
async def get_company_profile(user: dict = Depends(get_current_user)):
    """Fetches the authenticated recruiter's company profile from Firestore."""
    db = firebase_db()
    doc = db.collection("companies").document(user["uid"]).get()
    if not doc.exists:
        return {"profile": None}
    return {"profile": doc.to_dict()}
